pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go

```diff
@@ -1081,15 +1095,12 @@ func (h hooks) Filter(testContext Context) hooks {
 	return result
 }
 
-// AsSteps transforms the sequence of hooks into a corresponding test
-// step. If there is only one hook, the corresponding `runHookStep` is
-// returned. Otherwise, a `concurrentRunStep` is returned, where every
-// hook is run concurrently. `stopChans` should either be `nil` for
-// steps that are not meant to be run in the background, or contain
-// one stop channel (`shouldStop`) for each hook.
-func (h hooks) AsSteps(
-	label string, prng *rand.Rand, testContext *Context, stopChans []shouldStop, isLocal bool,
-) []testStep {
+// AsSteps transforms the sequence of hooks into a sequence of
+// `*singleStep` structs to be run in some way by the planner
+// (sequentially, concurrently, etc). `stopChans` should either be
+// `nil` for steps that are not meant to be run in the background, or
+// contain one stop channel (`shouldStop`) for each hook.
+func (h hooks) AsSteps(prng *rand.Rand, testContext *Context, stopChans []shouldStop) []testStep {
```

Comment:
Nice! I like that `concurrently` concern was moved to the caller(s).

Reply:
Yeah, having this function return a concurrent step was a mistake, it's the planner's responsibility to decide how to run steps.